librz/core/cmd_debug.c

```diff
@@ -1443,376 +1357,462 @@ static bool get_bin_info(RzCore *core, const char *file, ut64 baseaddr, PJ *pj,
+			rz_cons_println(pj_string(pj));
+			pj_free(pj);
+		}
 		free(ptr);
 	} break;
-	case ' ': // "dm "
+	case '.': // "dmi."
 	{
-		int size;
-		char *p = strchr(input + 2, ' ');
-		if (p) {
-			*p++ = 0;
-			addr = rz_num_math(core->num, input + 1);
-			size = rz_num_math(core->num, p);
-			rz_debug_map_alloc(core->dbg, addr, size, false);
-		} else {
-			eprintf("Usage: dm addr size\n");
-			return false;
+		map = get_closest_map(core, addr);
+		if (map) {
+			ut64 closest_addr = UT64_MAX;
+			RzList *symbols = rz_bin_get_symbols(core->bin);
+			RzBinSymbol *symbol, *closest_symbol = NULL;
+
+			rz_list_foreach (symbols, iter, symbol) {
+				if (symbol->vaddr > addr) {
+					if (symbol->vaddr - addr < closest_addr) {
+						closest_addr = symbol->vaddr - addr;
+						closest_symbol = symbol;
+					}
+				} else {
+					if (addr - symbol->vaddr < closest_addr) {
+						closest_addr = addr - symbol->vaddr;
+						closest_symbol = symbol;
+					}
+				}
+			}
+			if (closest_symbol) {
+				RzCoreBinFilter filter;
+				filter.offset = 0LL;
+				filter.name = (char *)closest_symbol->name;
+
+				rz_bin_set_baddr(core->bin, map->addr);
+				rz_core_bin_info(core, RZ_CORE_BIN_ACC_SYMBOLS, NULL, false, true, &filter, NULL);
+			}
 		}
 	} break;
-	case '-': // "dm-"
-		if (input[1] != ' ') {
-			eprintf("|ERROR| Usage: dm- [addr]\n");
-			break;
-		}
-		addr = rz_num_math(core->num, input + 2);
+	default:
+		rz_core_cmd_help(core, help_msg_dmi);
+		break;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dmp
+RZ_IPI RzCmdStatus rz_debug_memory_permission_handler(RzCore *core, int argc, const char **argv) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzListIter *iter;
+	RzDebugMap *map;
+	ut64 addr = 0, size = 0;
+	int perms;
+	if (argc == 3) { // dmp <size> <perms> @ <addr>
+		addr = core->offset;
+		size = rz_num_math(core->num, argv[1]);
+		perms = rz_str_rwx(argv[2]);
+		rz_debug_map_protect(core->dbg, addr, (int)size, perms);
+	} else if (argc == 2) { // dmp <perms>
+		addr = UT64_MAX;
 		rz_list_foreach (core->dbg->maps, iter, map) {
-			if (addr >= map->addr && addr < map->addr_end) {
-				rz_debug_map_dealloc(core->dbg, map);
-				rz_debug_map_sync(core->dbg);
-				return true;
+			if (core->offset >= map->addr && core->offset < map->addr_end) {
+				addr = map->addr;
+				size = map->size;
+				break;
 			}
 		}
-		eprintf("The address doesn't match with any map.\n");
-		break;
-	case 'L': // "dmL"
-	{
-		int size;
-		char *p = strchr(input + 2, ' ');
-		if (p) {
-			*p++ = 0;
-			addr = rz_num_math(core->num, input + 1);
-			size = rz_num_math(core->num, p);
-			rz_debug_map_alloc(core->dbg, addr, size, true);
+		perms = rz_str_rwx(argv[1]);
+		if (addr != UT64_MAX && perms >= 0) {
+			rz_debug_map_protect(core->dbg, addr, (int)size, perms);
 		} else {
-			eprintf("Usage: dmL addr size\n");
-			return false;
+			return RZ_CMD_STATUS_WRONG_ARGS;
```

Comment:
```suggestion
			return RZ_CMD_STATUS_ERROR;
```
WRONG_ARGS usually means you passed the wrong number of args, but in this case it is something else.